lib/liquid/standardfilters.rb

```diff
@@ -93,7 +94,13 @@ def truncatewords(input, words = 15, truncate_string = "...")
       words = Utils.to_integer(words)
       words = 1 if words <= 0
 
-      wordlist = input.split(" ", words + 1)
+      wordlist = begin
+        input.split(" ", words + 1)
+      rescue RangeError
+        raise if words + 1 < MAX_INT
+        # e.g. integer #{words} too big to convert to `int'
+        raise Liquid::ArgumentError, "integer #{words} too big for truncatewords"
```

Comment:
Should this be raised before calling `split`, right after assigning `words`?

Reply:
You mean make a change like

```diff
diff --git a/lib/liquid/standardfilters.rb b/lib/liquid/standardfilters.rb
index 150406a6..be16d494 100644
--- a/lib/liquid/standardfilters.rb
+++ b/lib/liquid/standardfilters.rb
@@ -94,13 +94,11 @@ module Liquid
       words = Utils.to_integer(words)
       words = 1 if words <= 0
 
-      wordlist = begin
-        input.split(" ", words + 1)
-      rescue RangeError
-        raise if words + 1 < MAX_INT
-        # e.g. integer #{words} too big to convert to `int'
+      if words + 1 >= MAX_INT
         raise Liquid::ArgumentError, "integer #{words} too big for truncatewords"
       end
+
+      wordlist = input.split(" ", words + 1)
       return input if wordlist.length <= words
 
       wordlist.pop
```

That should be functionally equivalent, however, from a performance perspective that adds extra overhead to the successful case.  In contrast, adding a rescue block doesn't add any extra overhead when an exception isn't raised, since MRI's exception handling is zero cost to the non-exception code path.

Reply:
Yeah that's what I meant. Thanks for the explanation! 👍